pkg/controllers/backup/controller.go

```diff
@@ -85,12 +89,24 @@ func Register(
 }
 
 func (h *handler) OnBackupChange(_ string, backup *v1.Backup) (*v1.Backup, error) {
+	var err error
+
+	if h.metricsServerEnabled {
+		backupList, err := h.backups.List(k8sv1.ListOptions{})
+		if err != nil {
+			logrus.Error("Error getting Backup CR list. Failed to update metrics.")
+			return nil, err
+		}
+
+		monitoring.UpdateBackupMetrics(backupList.Items)
+	}
```

Comment:
This should run as a background task, instead of when we enqueue a backup resource